### ISSUE TITLE
PETSc CI Undo Reversion

### ISSUE THREAD

**Issue by jeremylt** [CEED/libCEED#1686]
See discussion here https://github.com/CEED/libCEED/pull/1681

We temporarily reverted https://gitlab.com/petsc/petsc/-/commit/7ada2a5134eeb4113b6693316dd2660902f08a96

but we need to find the root cause of the bug and patch it, then revert this reversion.

**Comment by jeremylt** [comment]
Note: debugging here was not fruitful https://gitlab.com/petsc/petsc/-/merge_requests/7929

**Comment by zatkins-dev** [comment]
I think this is a fix: https://gitlab.com/petsc/petsc/-/merge_requests/9311

**Comment by jeremylt** [comment]
amazing - the fluids tests pass with that?

**Comment by zatkins-dev** [comment]
yep, I tested VECHIP/AIJHIPSPARSE specifically, which does have the error too (it was previously using Kokkos which doesn't, I'll upload the MR for that fix tomorrow.) I'll also double check CUDA, I was feeling lazy at 5:30 and didn't want to rebuild PETSc

**Comment by jeremylt** [comment]
Fix merged into PETSc main and release
